tst/com/amazon/corretto/crypto/provider/test/EdDSATest.java

```diff
@@ -0,0 +1,284 @@
+    nativeGen = null;
+    jceGen = null;
+    bcGen = null;
+  }
+
+  @Test
+  public void uniqueKeyGen() {
+    final KeyPair kp1 = nativeGen.generateKeyPair();
+    final KeyPair kp2 = nativeGen.generateKeyPair();
+
+    final byte[] pk1 = kp1.getPrivate().getEncoded();
+    final byte[] pk2 = kp2.getPrivate().getEncoded();
+
+    final byte[] pbk1 = kp1.getPublic().getEncoded();
+    final byte[] pbk2 = kp2.getPublic().getEncoded();
+
+    assertTrue(!Arrays.equals(pk1, pk2));
+    assertTrue(!Arrays.equals(pbk1, pbk2));
+  }
+
+  @Test
+  public void keyGenValidation() throws GeneralSecurityException {
+    // Generate Keys with ACCP & Sign/Verify with SunEC
+    final byte[] message = new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    final KeyPair keyPair = nativeGen.generateKeyPair();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 =
+        new PKCS8EncodedKeySpec(keyPair.getPrivate().getEncoded());
+    final X509EncodedKeySpec publicKeyX509 =
+        new X509EncodedKeySpec(keyPair.getPublic().getEncoded());
+
+    final KeyFactory kf = KeyFactory.getInstance("EdDSA", "SunEC");
+
+    final PrivateKey privateKey = kf.generatePrivate(privateKeyPkcs8);
+    final PublicKey publicKey = kf.generatePublic(publicKeyX509);
+
+    final Signature eddsa = Signature.getInstance("Ed25519", "SunEC");
+
+    eddsa.initSign(privateKey);
+    eddsa.update(message, 0, message.length);
+    final byte[] signature = eddsa.sign();
+    eddsa.initVerify(publicKey);
+    eddsa.update(message);
+    assertTrue(eddsa.verify(signature));
+  }
+
+  @Test
+  public void keyFactoryValidation() throws GeneralSecurityException {
+    final KeyPair keyPair = jceGen.generateKeyPair();
+
+    final byte[] privateKeyJCE = keyPair.getPrivate().getEncoded();
+    final byte[] publicKeyJCE = keyPair.getPublic().getEncoded();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 = new PKCS8EncodedKeySpec(privateKeyJCE);
+    final X509EncodedKeySpec publicKeyX509 = new X509EncodedKeySpec(publicKeyJCE);
+
+    final KeyFactory kf = KeyFactory.getInstance("Ed25519", NATIVE_PROVIDER);
+
+    final byte[] privateKeyACCP = kf.generatePrivate(privateKeyPkcs8).getEncoded();
+    final byte[] publicKeyACCP = kf.generatePublic(publicKeyX509).getEncoded();
+
+    // Confirm that ACCP & SunEC keys are equivalent
+    assertTrue(Arrays.equals(privateKeyACCP, privateKeyJCE));
+    assertTrue(Arrays.equals(publicKeyACCP, publicKeyJCE));
+  }
+
+  @Test
+  public void jceInteropValidation() throws GeneralSecurityException {
+    // Generate keys with ACCP and use JCE KeyFactory to get equivalent JCE Keys
+    final KeyPair keyPair = nativeGen.generateKeyPair();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 =
+        new PKCS8EncodedKeySpec(keyPair.getPrivate().getEncoded());
+    final X509EncodedKeySpec publicKeyX509 =
+        new X509EncodedKeySpec(keyPair.getPublic().getEncoded());
+
+    final KeyFactory kf = KeyFactory.getInstance("Ed25519", "SunEC");
```

Comment:
Without passing the keys to a keyfactory, one should be able to use `KeyPair` for signing and verifying.

Reply:
Since the ACCP keys don't use the specific Ed classes from Java, we can't directly use them for sign/verify. Using KeyFactory is the current fix, but being able to import the Ed classes from JDK 15+ would allow full compatibility. That wasn't possible currently, since ACCP needs to compile on jdk 8.

Reply:
SunEC's EdDSA KeyFactory was added in JDK 15. I don't think I could include it in our key generation, since it would cause the same compilation issues.

Reply:
I don't think you even need to check JDK version. All you need is to see if `KeyFactory.getInstance("Ed25519")` doesn't throw an exception; if not, then it's a JDK version that supports Ed25519 keys.

Reply:
Updated, but when running the benchmark with the new changes, noticed ~20% dropoff in performance for key pair generation. (160,000 ops/second --> 126,000 ops/second). Do we think that tradeoff is worth it for easier interoperability?